r-publishing.Rmd

```diff
@@ -0,0 +1,306 @@
+
+To add a link to `report.html` in `index.Rmd` I would add a line like:
+
+```markdown
+See the [full report](report.html)
+```
+
+You should use relative URLs to reference any of **your** files (i.e. those in `docs/`). That way if you ever rename your repository, move it, or use a different hosting platform, your links will all work without changes.  You must use absolute links for files that reside elsewhere on the internet.
+
+### Exercise: Relative links
+
+Imagine your `docs/` folder had the following structure:
+
+```
+├── docs
+│   ├── index.Rmd
+│   ├── index.html
+│   ├── diagrams
+│   │   └── workflow.png
+|   ├── reports
+|   │   ├── jan.Rmd
+|   │   ├── jan.html
+|   │   ├── feb.Rmd
+|   │   └── feb.html
+└── sharing-work.Rproj
+```
+
+Using a relative URL, how would you refer to:
+
+* `jan.html` from `index.html`?
+* `feb.html` from `jan.html`?
+* `workflow.png` from `index.html`?
+
+### Exercise: Add links to `index.html`
+
+Add to `index.Rmd`:
+
+* a link to `report.html` using a relative link, and
+* a link your GitHub repository using an absolute link.
+
+### Linking to sections within a page
+
+URLs can also refer to places inside the current page, most usually to another section.  In Rmarkdown you've seen you create headings using `#`.  For example, an Appendix subsection might be:
+```markdown
+## Appendix
+```
+
+If we want to link to this section from elsewhere, you  proceed the section name with `#` in the URL.  For example, if this section is in `report.Rmd` and you want to link to it elsewhere in `report.Rmd`, you could use:
+```markdown
+See more details in the [appendix](#appendix)
+```
+
+The URL `#appendix` is interpreted as the heading with ID `appendix` in the current page.  Rmarkdown creates IDs for all sections (and subsections) automatically, by converting to lower case and replacing spaces with dashes (`-`).  But, you can explicitly set IDs too, by adding the ID with the `#` inside curly braces after the section heading.  For instance you might prefer the shorter `appen` ID.  You need to set it where the heading occurs:
+```markdown
+## Appendix {#appen}
+```
+
+Then you can link to it using this shorter ID elsewhere:
+```markdown
+See more details in the [appendix](#appen)
+```
+
+You can also use this strategy to link to sections in other pages by including the relative URL first.  For instance, to refer to this "Appendix" section from `index.html` you could include in `index.Rmd`:
+```markdown
+Some gory details of the analysis can also be found in the [Appendix of the report](report.html#appen)
+```
+
+### Exercise: Add and link to a section in `report.Rmd`
+
+Add a new section to `report.Rmd` and include a link to it in `index.Rmd`.  You'll need to Knit both `report.Rmd` and `index.Rmd`, and commit and push the HTML files to check your work.
+
+### Including images
```

Comment:
I think it would be useful to include how to resize images with `{}` in this section.

```
[A picture of me](images/me.png){width=50} # default unit is px
[A picture of me](images/me.png){width=50%}
```

Reply:
This is new to me!  I'll add.